lib/identify.c

```diff
@@ -177,28 +177,28 @@ static void flood_fill_seed(struct quirc *q, int x, int y, int from, int to,
 
 static uint8_t otsu(const struct quirc *q)
 {
-	int numPixels = q->w * q->h;
+	unsigned int numPixels = q->w * q->h;
 
 	// Calculate histogram
 	unsigned int histogram[UINT8_MAX + 1];
```

Comment:
i'm not sure what you mean here. if you make the change, you should change the code below as well because the last element of the array is actually used. if you want to change the code as well, i guess it should be a dedicated PR.

Reply:
My mistake, I thought the array was off-by-one too large but after your comment and re-reading it's actually the correct size indeed.